test/IDE/complete_ambiguous.swift

```diff
@@ -117,13 +119,33 @@ let x: A = overloadedReturn(1).#^RELATED_EXTRAARG^#
 // RELATED-DAG: Decl[InstanceMethod]/CurrNominal/TypeRelation[Invalid]: doBThings()[#Void#]{{; name=.+$}}
 // RELATED: End completions
 
-func takesA(_ callback: () -> A) -> B {}
+func takesClosureGivingA(_ callback: () -> A) -> B {}
 func takesB(_ item: B) {}
 
-takesB((takesA { return overloadedReturn().#^RELATED_INERROREXPR^# }).)
+takesB((takesClosureGivingA { return overloadedReturn().#^RELATED_INERROREXPR^# }).)
+
+func overloadedArg(_ arg: A) -> A {}
+func overloadedArg(_ arg: B) -> B {}
+
+overloadedArg(.#^UNRESOLVED_AMBIGUOUS^#)
+
+// UNRESOLVED_AMBIGUOUS: Begin completions, 4 items
+// UNRESOLVED_AMBIGUOUS-DAG: Decl[InstanceMethod]/CurrNominal: doAThings({#(self): A#})[#() -> A#]{{; name=.+$}}
+// UNRESOLVED_AMBIGUOUS-DAG: Decl[Constructor]/CurrNominal/TypeRelation[Identical]: init()[#A#]{{; name=.+$}}
+// UNRESOLVED_AMBIGUOUS-DAG: Decl[InstanceMethod]/CurrNominal/TypeRelation[Invalid]: doBThings({#(self): B#})[#() -> Void#]{{; name=.+$}}
+// UNRESOLVED_AMBIGUOUS-DAG: Decl[Constructor]/CurrNominal/TypeRelation[Identical]: init()[#B#]{{; name=.+$}}
+// UNRESOLVED_AMBIGUOUS: End completions
+
+takesB(overloadedArg(.#^UNRESOLVED_UNAMBIGUOUS^#)
+
+// UNRESOLVED_UNAMBIGUOUS: Begin completions, 2 items
+// UNRESOLVED_UNAMBIGUOUS-DAG: Decl[InstanceMethod]/CurrNominal/TypeRelation[Invalid]: doBThings({#(self): B#})[#() -> Void#]{{; name=.+$}}
+// UNRESOLVED_UNAMBIGUOUS-DAG: Decl[Constructor]/CurrNominal/TypeRelation[Identical]: init()[#B#]{{; name=.+$}}
+// UNRESOLVED_UNAMBIGUOUS: End completions
```

Comment:
I just realized this is probably undesirable behavior on the part of the filtering logic 😞. We get a viable solution choosing the `overloadedArg(_: B) -> B` overload and an unviable one choosing `overloadedArg(_: A) -> A` (because of arg/param mismatch) that gets filtered out. For code completions purposes we should probably ignore arg/param mismatches when the arg contains the code completion loc as it may be incomplete. E.g. the user may intend to do `overloadedArg(.somethingThatReturnsA()).somethingThatReturnsB()` making the `overloadedArg(_: A) -> A` overload viable as an arg for `takesB` again.

Reply:
Fixed.